### ISSUE TITLE
geth --exec exits cleanly even when supplied command is invalid

### ISSUE THREAD

**Issue by hiddentao** [ethereum/go-ethereum#2470]
#### System information

Geth version: 1.3.5
OS & Version: OSX
#### Expected behaviour

If I run `geth --exec "invalid()" attach ipc://.../geth.ipc` it should exit with a non-zero error code.
#### Actual behaviour

It exits with error code 0 (even though the above command is clearly invalid) and the error message returned in `stdout` rather than `stderr`:

``` bash
$ geth --exec "invalid()" attach ipc:///var/folders/4v/br6x6mlx113235v1lz39nwfc0000gn/T/tmp-74391lqz0pEtM1uNl/geth.ipc
error: ReferenceError: 'invalid' is not defined
$ echo $?
0
```
#### Steps to reproduce the behaviour

Run `geth --exec "invalid()" attach ipc://.../geth.ipc`  (replace `ipc..` with valid path)

**Comment by tgerring** [comment]
The JSRE correctly evaluated the exec statement without error (even though there is an error in the JavaScript)

Should a VM containing another VM bubble up its error code or what error code would you expect to see?

**Comment by fjl** [comment]
I think it's a reasonable request.

**Comment by hiddentao** [comment]
Just want to add, this is useful for other scripts which may call geth - it would be better to be able to rely on the exit code rather than have to scan the logs.

**Comment by fjl** [comment]
This is resolved in 1.5.0 AFAIK.

**Comment by tuxofil** [comment]
Oops. Regression?
```
$ geth version
Geth
Version: 1.8.3-unstable
Architecture: amd64
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.10
Operating System: linux
GOPATH=
GOROOT=/build/ethereum-1.8.2/go

$ geth attach --exec 'invalid();' data/geth.ipc && echo $?
ReferenceError: 'invalid' is not defined
    at <anonymous>:1:1
0
```
